tests/apps/test_data.py

```diff
@@ -328,3 +332,111 @@ def test_data_dim_correct(self, dataset_patched):
         assert m == dataset_patched.n_features
         assert p == len(dataset_patched.adjs)
         assert q == len(dataset_patched.unit_data)
+
+
+@pytest.mark.parametrize("datasets", WATER_DATASETS_LIST)
+class TestWaterDatasets:
+    """Tests for the ``Water`` class"""
+
+    @pytest.fixture
+    def dataset(self, datasets):
+        """Fixture for loading each of the datasets in ``WATER_DATASETS_LIST``"""
+        yield datasets(t=0)
+
+    def test_w_dims(self, dataset):
+        """Test if w has correct shape"""
+        w = dataset.w
+        assert w.shape == (dataset.modes,)
+
+    def test_u_dims(self, dataset):
+        """Test if U has correct shape"""
+        U = dataset.U
+        assert U.shape == (dataset.modes, dataset.modes)
+
+    def test_u_unitary(self, dataset):
+        """Test if U is unitary"""
+        U = dataset.U
+        assert np.allclose(np.eye(U.shape[0]), U.conj().T @ U)
+
+    def test_sample_dims(self, dataset):
+        """Test if sample has correct shape"""
+        assert np.shape(dataset[:]) == (5000, dataset.modes)
+
+    def test_times_water(self, dataset):
+        """Test if available times are correct"""
+        t = dataset.available_times()
+        assert np.allclose(t, np.linspace(0, 270, 28))
+
+    def test_dataset_water(self, datasets):
+        """Test if function raises a ``ValueError`` when an incorrect time is given"""
+        with pytest.raises(ValueError, match="The selected time is not correct"):
+            datasets(t=7)
+
+
+@pytest.mark.parametrize("datasets", PYRROLE_DATASETS_LIST)
+class TestPyrroleDatasets:
+    """Tests for the ``Pyrrole`` class"""
+
+    @pytest.fixture
+    def dataset(self, datasets):
+        """Fixture for loading each of the datasets in ``Pyrrole_DATASETS_LIST``"""
+        yield datasets(t=0)
+
+    def test_ri_dims(self, dataset):
+        """Test if ri has correct shape"""
+        ri = dataset.ri
+        assert ri.shape == (30,)

```

Comment:
Where does the 30 come from :thinking:

Reply:
`ri` are the molecular coordinates. pyrrole has 10 atoms. So I believe this is a flattened version of the three-dimensional coordinates of the 10 atoms?